### PR TITLE
C359001 Error after typing non existing item location form in Bulk Edit (firebird) (TaaS)

### PR DESCRIPTION
**Pull request by kuchmenkoepam**
[359001](https://foliotest.testrail.io/index.php?/cases/view/359001)
[Allure report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5977/allure/)
image

### CODE REVIEW

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js [hunk 1]
```diff
@@ -38,6 +39,7 @@ const confirmChangesButton = Button('Confirm changes');
 const downloadChnagedRecordsButton = Button('Download changed records (CSV)');
 const bulkEditFirstRow = RepeatableFieldItem({ index: 0 });
 const bulkEditSecondRow = RepeatableFieldItem({ index: 1 });
+const searchSection = Section({ id: 'sl-container-stripes-selection-41' });
```
Comment (manowarqqq) [inline cypress/support/fragments/bulk-edit/bulk-edit-actions.js:42]: Is there any filter  to this section? this id doesn't look stable

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js [hunk 2]
```diff
@@ -922,4 +936,9 @@ export default {
       Option({ value: 'SUPPRESS_FROM_DISCOVERY' }).exists(),
     ]);
   },
+
+  fillLocation(location) {
+    cy.do(Button('Select control\nSelect location').click());
```
Comment (manowarqqq) [inline cypress/support/fragments/bulk-edit/bulk-edit-actions.js:941]: Please, move this button to variables

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js [hunk 3]
```diff
@@ -922,4 +936,9 @@ export default {
...
+    cy.get('input[aria-label=" options filter"]').type(location).type('{enter}');
```
Comment (manowarqqq) [inline cypress/support/fragments/bulk-edit/bulk-edit-actions.js:942]: Can be done with SelectionList interactor
const locationList = SelectionList(...)
cy.do(locationList.filter(location))

File: cypress/support/fragments/bulk-edit/bulk-edit-search-pane.js [hunk 4]
```diff
@@ -505,6 +505,10 @@ export default {
     cy.expect(HTML(name).exists());
   },
 
+  verifyInputLabelAbsent(name) {
```
Comment (manowarqqq) [inline cypress/support/fragments/bulk-edit/bulk-edit-search-pane.js:508]: the same. Make one method for absent and exists

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js [hunk 5]
```diff
@@ -98,6 +100,30 @@ export default {
     cy.expect([plusBtn.exists(), Button({ icon: 'trash', disabled: true }).exists()]);
   },
 
+  verifyOptionsDropdown(isExist = true) {
+    if (isExist) {
+      cy.expect(bulkPageSelections.valueType.exists());
+    } else {
+      cy.expect(bulkPageSelections.valueType.exists()).absent();
```
Comment (manowarqqq) [inline cypress/support/fragments/bulk-edit/bulk-edit-actions.js:107]: No need exists, only absent
  cy.expect(bulkPageSelections.valueType.absent()).;

**Review by manowarqqq — APPROVED** [review]

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js [hunk 6]
```diff
@@ -98,6 +100,30 @@ export default {
...
 
+  verifyOptionsDropdown(isExist = true) {
+    if (isExist) {
+      cy.expect(bulkPageSelections.valueType.exists());
+    } else {
+      cy.expect(bulkPageSelections.valueType.absent());
+    }
+  },
+
+  verifyActionDropdown(isExist = true) {
```
Comment (nayimovag) [inline cypress/support/fragments/bulk-edit/bulk-edit-actions.js:111]: duplication of `isSelectActionAbsent`

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-error-after-typing-non-existing-item-location.cy.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+let user;
+const item = {
+  barcode: getRandomPostfix(),
+  instanceName: `instance-${getRandomPostfix()}`,
+};
+const itemBarcodesFileName = `itemBarcodes_${getRandomPostfix()}.csv`;
+
+describe('Bulk Edits', () => {
+  describe('Bulk Edit - Items', () => {
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-error-after-typing-non-existing-item-location.cy.js:18]: please change to 
`describe('bulk-edit', () => {
  describe('in-app approach', () => {`
so the tests are sorted correctly in the allure

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js [hunk 8]
```diff
@@ -665,6 +691,11 @@ export default {
   isCommitButtonDisabled(isDisabled) {
     cy.expect(commitChanges.has({ disabled: isDisabled }));
   },
+
+  isConfirmChangesButtonDisabled(isDisabled) {
```
[the remark this answers is not in the document]
Comment (kuchmenkoepam) [reply cypress/support/fragments/bulk-edit/bulk-edit-actions.js]: fixed

File: cypress/support/fragments/bulk-edit/bulk-edit-actions.js (hunk 6 above)
Comment (kuchmenkoepam) [reply cypress/support/fragments/bulk-edit/bulk-edit-actions.js]: deleted

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-error-after-typing-non-existing-item-location.cy.js (hunk 7 above)
Comment (kuchmenkoepam) [reply cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-error-after-typing-non-existing-item-location.cy.js]: fixed

**Review by nayimovag — APPROVED** [review]

**Pull request merged**